Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.cpp

```diff
@@ -111,11 +111,8 @@ void GraphicsLayerCoordinated::setNeedsDisplayInRect(const FloatRect& initialRec
 
     addRepaintRect(rect);
 
-    if (!m_dirtyRegion) {
-        static constexpr FloatSize minDamageSize = { 512, 512 };
-        auto viewVisibleSize = client().enclosingFrameViewVisibleSize();
-        m_dirtyRegion = Damage(m_size.constrainedBetween(minDamageSize, viewVisibleSize));
-    }
+    if (!m_dirtyRegion)
+        m_dirtyRegion = Damage(m_size, Damage::Mode::Rectangles, 32);
```

Comment:
I'm not sure about 32 here - doesn't this mean that for 32x32 layer we get `Damage` with up to 32 rects?

Reply:
This is what we had before https://commits.webkit.org/271045@main and also what Apple does. So this is basically a revert of that patch but using Damage so that we unite in a better way instead of always in the first rectangle.